### ISSUE TITLE
Interactive elements should have unique accessible names

### ISSUE THREAD

**Issue by dengeist** [morgannewman/womby#7]
**Affected component:** `NoteCard`

**Problem:** All delete buttons in the sidebar component will announce the same name: `Delete`. Because low-vision and blind users do not get the meaning of interactive elements from their context in the way that sighted users do, interactive elements should have accessible names that are unique and  self-sufficiently descriptive,

**Proposed solution:** Use `aria-label` on the delete buttons, or hidden text within the buttons, to create distinction for users listening to the app. Users would benefit from hearing _which_ note will be deleted. Ex: "Delete TITLE_OF_NOTE".

**Relevant WCAG criteria:**
- [2.4.6 Headings and Labels - Level AA](https://www.w3.org/WAI/WCAG21/quickref/#headings-and-labels)

**Resources:**
- [Invisible content just for screen reader users](https://webaim.org/techniques/css/invisiblecontent/)
- [What is an accessible name?](https://developer.paciellogroup.com/blog/2017/04/what-is-an-accessible-name/)
